exp/controllers/azuremangedcontrolplane_controller.go

```diff
@@ -24,6 +24,8 @@ import (
 	"github.com/pkg/errors"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/cluster-api/util/annotations"
+
 	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1alpha3"
 	infrav1exp "sigs.k8s.io/cluster-api-provider-azure/exp/api/v1alpha3"
```

Comment:
SetupWithManager should be updated to:
- Add `WithEventFilter(predicates.ResourceNotPaused(r.Log)).` to the `ctrl.NewControllerManagedBy()` chain to filter out reconciliations when the paused annotation is set on any of the resources being watched.
- Add a Watch for `Cluster` that uses `predicates.ClusterUnpausedAndInfrastructureReady()` to properly trigger reconciliation when Cluster infrastructure becomes ready and when the Cluster is unpaused, similar to what is being done here: https://github.com/kubernetes-sigs/cluster-api/blob/2bb56ed2eb2a91cb59b740cb286992952da3dfed/bootstrap/kubeadm/controllers/kubeadmconfig_controller.go#L121-L130